File: 7KYU/Functions of Integers on Cartesian Plane/solution.clj
(ns funcij.core)

(defn sumin [n]
  (loop [acc 0 i 1 n n]
    (if (zero? n)
      acc
      (recur (+ acc (* i n)) (+ 2 i) (dec n)))))

(defn sumax [n]
  (loop [acc 0
         i (dec (* n 2))
         n n]
    (if (neg? i)
      acc
      (recur (+ acc (* i n)) (- i 2) (dec n)))))

(defn sumsum [n]
  (+ (sumax n) (sumin n)))

File: 7KYU/Functions of Integers on Cartesian Plane/solution.go
package kata

func SuMin(m int) int64 {
	acc := int64(0)
	i := int64(1)
	for m > 0 {
		acc = acc + i*int64(m)
		i += 2
		m--
	}
	return acc
}

func SuMax(m int) int64 {
	acc := int64(0)
	i := int64((m * 2) - 1)

	for m >= 0 {
		acc = acc + i*int64(m)
		i -= 2
		m--
	}
	return acc
}

func SumSum(m int) int64 {
	return SuMax(m) + SuMin(m)
}
